File: back/src/main/java/com/quest/buddy/models/Event.java
package com.quest.buddy.models;

import org.modelmapper.ModelMapper;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

import javax.persistence.*;

import com.quest.buddy.dtos.EventDto;

@Entity
@Table(name = "event")
public class Event implements BaseModel<EventDto> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", length = 80, nullable = false)
    private String title;

    @Column(name = "urlImage", length = 200)
    private String urlImage;

    @Column(name = "nbPartnerMax")
    private int nbPartnerMax;

    @Column(name = "dateEventCreate")
    private LocalDateTime dateEventCreate;

    @Column(name = "dateEventModify")
    private LocalDateTime dateEventModify;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Column(name = "startTime")
    private LocalDateTime startTime;

    @Column(name = "endTime")
    private LocalDateTime endTime;

    @Column(name = "place")
    private String place;

    @Column(name = "flatGender")
    private byte flatGender;

    @Column(name = "ageMax")
    private byte ageMax;

    @Column(name = "ageMin")
    private byte ageMin;

    @Column(name = "description" , length = 300)
    private String description;

    @Column(name = "state")
    private byte state;

    @ManyToOne
    @JoinColumn(name = "userId", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "sportId", referencedColumnName = "id")
    private Sport sport;

    @ManyToOne
    @JoinColumn(name = "localisation_Id", referencedColumnName = "id")
    private Localisation localisation;

    public Event() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }

    public int getNbPartnerMax() {
        return nbPartnerMax;
    }

    public void setNbPartnerMax(int nbPartnerMax) {
        this.nbPartnerMax = nbPartnerMax;
    }

    public Sport getSport() {
        return sport;
    }

    public void setSport(Sport sport) {
        this.sport = sport;
    }

    public LocalDateTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalDateTime startTime) {
        this.startTime = startTime;
    }

    public LocalDateTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalDateTime endTime) {
        this.endTime = endTime;
    }

    public String getPlace() {
        return place;
    }

    public void setPlace(String place) {
        this.place = place;
    }

    public byte getFlatGender() {
        return flatGender;
    }

    public void setFlatGender(byte flatGender) {
        this.flatGender = flatGender;
    }

    public byte getAgeMax() {
        return ageMax;
    }

    public void setAgeMax(byte ageMax) {
        this.ageMax = ageMax;
    }

    public byte getAgeMin() {
        return ageMin;
    }

    public void setAgeMin(byte ageMin) {
        this.ageMin = ageMin;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public byte getState() {
        return state;
    }

    public void setState(byte state) {
        this.state = state;
    }

    public Localisation getLocalisation() {
        return localisation;
    }

    public void setLocalisation(Localisation localisation) {
        this.localisation = localisation;
    }

    public LocalDateTime getDateEventCreate() {
        return dateEventCreate;
    }

    public void setDateEventCreate(LocalDateTime dateEventCreate) {
        this.dateEventCreate = dateEventCreate;
    }

    public LocalDateTime getDateEventModify() {
        return dateEventModify;
    }

    public void setDateEventModify(LocalDateTime dateEventModify) {
        this.dateEventModify = dateEventModify;
    }

    public String getInfoDateEventCreate(){
        //create if par today
        return "Publié - " + this.dateEventCreate.toLocalDate().toString() + " "+this.dateEventCreate.toLocalTime().toString() ;
    }
    public String getInfoDateEvent(){
        //15 July 2021        10h00 ->11h30
        String dateFormatStart,dateFormatEnd ;
        dateFormatStart =this.startTime.toLocalDate().toString() +" " + this.startTime.toLocalTime().toString() ;
        dateFormatEnd=this.endTime.toLocalDate().toString() +" "+this.endTime.toLocalTime().toString();
        return  dateFormatStart +" --> "+dateFormatEnd;
    }
    public String getInfoNroPartner(){
        return this.nbPartnerMax + " Partners";
    }
    @Override
    public EventDto toDto(){
        ModelMapper modelMapper = new ModelMapper();
        EventDto eventDto = modelMapper.map(this, EventDto.class);
        return eventDto;
    }


}

File: front/src/app/models/event.ts

import {User} from "../components/users/user";
import {Sport} from "./sport";
import {Localisation} from "./localisation";

export class Event{
  //constructor(
    private _title: string | undefined;
    private _urlImage:string| undefined;
    private _nbPartnerMax:string| undefined;
    private _sport: Sport| undefined;
    private _dateEventCreate:string| undefined;
    private _startTime:string| undefined;
    private _endTime:string| undefined;
    private _place:string| undefined;
    private _flatGender:boolean| undefined;
    private _ageMax:string| undefined;
    private _ageMin:string| undefined;
    private _user: User| undefined;
    private _description:string| undefined;
    private _state: number| undefined;
    private _localisation:Localisation| undefined;
  //){}


  get title(): string | undefined {
    return this._title;
  }

  set title(value: string | undefined) {
    this._title = value;
  }

  get urlImage(): string | undefined {
    return this._urlImage;
  }

  set urlImage(value: string | undefined) {
    this._urlImage = value;
  }

  get nbPartnerMax(): string | undefined {
    return this._nbPartnerMax;
  }

  set nbPartnerMax(value: string | undefined) {
    this._nbPartnerMax = value;
  }

  get sport(): Sport | undefined {
    return this._sport;
  }

  set sport(value: Sport | undefined) {
    this._sport = value;
  }

  get dateEventCreate(): string | undefined {
    return this._dateEventCreate;
  }

  set dateEventCreate(value: string | undefined) {
    this._dateEventCreate = value;
  }

  get startTime(): string | undefined {
    return this._startTime;
  }

  set startTime(value: string | undefined) {
    this._startTime = value;
  }

  get endTime(): string | undefined {
    return this._endTime;
  }

  set endTime(value: string | undefined) {
    this._endTime = value;
  }

  get place(): string | undefined {
    return this._place;
  }

  set place(value: string | undefined) {
    this._place = value;
  }

  get flatGender(): boolean | undefined {
    return this._flatGender;
  }

  set flatGender(value: boolean | undefined) {
    this._flatGender = value;
  }

  get ageMax(): string | undefined {
    return this._ageMax;
  }

  set ageMax(value: string | undefined) {
    this._ageMax = value;
  }

  get ageMin(): string | undefined {
    return this._ageMin;
  }

  set ageMin(value: string | undefined) {
    this._ageMin = value;
  }

  get user(): User | undefined {
    return this._user;
  }

  set user(value: User | undefined) {
    this._user = value;
  }

  get description(): string | undefined {
    return this._description;
  }

  set description(value: string | undefined) {
    this._description = value;
  }

  get state(): number | undefined {
    return this._state;
  }

  set state(value: number | undefined) {
    this._state = value;
  }

  get localisation(): Localisation | undefined {
    return this._localisation;
  }

  set localisation(value: Localisation | undefined) {
    this._localisation = value;
  }

  get infoDateEventCreate(){
    //create if par today
    return this.dateEventCreate ;
}
get infoDateEvent(){
    //15 July 2021        10h00 ->11h30
    /*String dateFormatStart,dateFormatEnd ;
    dateFormatStart =this.startTime.toLocalDate().toString() +" " + this.startTime.toLocalTime().toString() ;
    dateFormatEnd=this.endTime.toLocalDate().toString() +" "+this.endTime.toLocalTime().toString();*/
    return  this.startTime;
}
get infoNroPartner(){
    return this.nbPartnerMax + " Partners";
}
}

